File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.py
class Solution(object):
    def reverseWords(self, s):
        """
        :type s: str
        :rtype: str
        """
        lst = s.split(" ")
        lst1 = [i for i in lst if i.strip()]
        lst2 = lst1[::-1]
        return " ".join(lst2)
        

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
import java.util.*;
class Solution {
    public String reverseWords(String s) {
        s = s.trim();
        String[] parts = s.split("\\s+");
        
        int i=0,j=parts.length-1;
        
        while(i < j){
            String temp = parts[i];
            parts[i] = parts[j];
            parts[j] = temp;
            i++;
            j--;
        }
        return String.join(" ",parts);
    }
}